process/txcost/gasUsed.go

```diff
@@ -7,34 +7,33 @@ import (
 )
 
 func (tcp *transactionCostProcessor) prepareGasUsed(senderShardID, receiverShardID uint32, res *data.TxCostResponseData) {
-	numResponses := len(tcp.responses)
 	extra := 0
 	if senderShardID != receiverShardID {
 		extra = 1
 	}
 
-	syncChan := make(chan struct{})
-	go func(c chan struct{}) {
-		defer func() {
-			if r := recover(); r != nil {
-				log.Warn("transactionCostProcessor.prepareGasUsed()", "stack", string(debug.Stack()))
+	tcp.computeResponsesGasUsed(extra, res)
+}
 
-				res.RetMessage = "something went wrong"
-				res.TxCost = 0
-			}
+func (tcp *transactionCostProcessor) computeResponsesGasUsed(extra int, res *data.TxCostResponseData) {
+	numResponses := len(tcp.responses)
 
-			c <- struct{}{}
-		}()
+	defer func() {
+		r := recover()
```

Comment:
It became clear only now: we usually recover from panics when using some 3rd party libraries that can panic. I suggest to remove this defer function and repair this function if it panics or add checks for wrongly provided params.

Reply:
refactored this part.